Add release notes for 6.0 (#2625)
sensu/sensu-docs: content/sensu-go/6.0/release-notes.md

---

## 6.0.0 release notes

**July 23, 2020** &mdash; The latest release of Sensu Go, version 6.0.0, is now available for download.

**PLACEHOLDER FOR RELEASE SYNOPSIS**

See the [upgrade guide][1] to upgrade Sensu to version 6.0.0.

**NEW FEATURES:**

- ([Commercial feature][162]) PLACEHOLDER.
- Added supported packages for Raspbian 7, 8, 9, and 10.
- Added binary-only distributions for FreeBSD `armv5`, `armv6`, and `armv7` and Linux `ppc64le` and `s390x`.

**IMPROVEMENTS:**

> - Added support for the `memberOf` attribute in Active Directory (AD).

Author: ```suggestion
- Added support for the `memberOf` attribute in Active Directory (AD).
- Output of `sensuctl asset add` now includes help for using the runtime asset.
```

**PLACEHOLDER FOR RELEASE SYNOPSIS**

See the [upgrade guide][1] to upgrade Sensu to version 6.0.0.

**BREAKING CHANGES FOR SENSU 6.0:**

- The database schema for entities has changed.
As a result, after upgrading to Sensu 6.0, you will not be able to use your database with older versions of Sensu.
- In [binary-only distributions][164], the web UI is now a standalone product that is no longer distributed inside the `sensu-backend` binary.
See the [Sensu Go Web repository][163] for more information.
- After initial creation, you cannot change your `sensu-agent` entity configuration by modifying the agent's configuration file.

**NEW FEATURES:**

- ([Commercial feature][162]) Sensu now logs a warning when secrets cannot be sent to an agent because mTLS is not enabled.
- ([Commercial feature][162]) Added `sensu.FetchEvent(entity, check)` and `sensu.ListEvents()` to the JavaScript filter execution environment so you can now query the Sensu event store for other events within the filter namespace.
- Added supported packages for Raspbian 7, 8, 9, and 10.
See the [supported platforms][165] page for a complete list of Sensu’s supported platforms.

> - Added [binary-only distributions][164] for FreeBSD `armv5`, `armv6`, and `armv7` and Linux `ppc64le` and `s390x`.

Author: ```suggestion
- Added [binary-only distributions][164] for FreeBSD `armv5`, `armv6`, and `armv7` and Linux `ppc64le` and `s390x`.
- Added the `is_silenced` Boolean attribute to the event.Check object to indicate whether the event was silenced at the time it was processed.
```

---

## 6.0.0 release notes

**July 23, 2020** &mdash; The latest release of Sensu Go, version 6.0.0, is now available for download.

**PLACEHOLDER FOR RELEASE SYNOPSIS**

See the [upgrade guide][1] to upgrade Sensu to version 6.0.0.

**BREAKING CHANGES FOR SENSU 6.0:**

- The database schema for entities has changed.
As a result, after upgrading to Sensu 6.0, you will not be able to use your database with older versions of Sensu.
- In [binary-only distributions][164], the web UI is now a standalone product that is no longer distributed inside the `sensu-backend` binary.
See the [Sensu Go Web repository][163] for more information.
- After initial creation, you cannot change your `sensu-agent` entity configuration by modifying the agent's configuration file.

**NEW FEATURES:**

- ([Commercial feature][162]) Sensu now logs a warning when secrets cannot be sent to an agent because mTLS is not enabled.
- ([Commercial feature][162]) Added `sensu.FetchEvent(entity, check)` and `sensu.ListEvents()` to the JavaScript filter execution environment so you can now query the Sensu event store for other events within the filter namespace.

> - Added supported packages for Raspbian 7, 8, 9, and 10.

Author: ```suggestion
- Added Linux packages for 386; armv5, armv6, and armv7; MIPS hard float, MIPS LE hard float, and MIPS 64 LE hard float; ppc64le; and s390x architectures.
```

See the [supported platforms][165] page for a complete list of Sensu’s supported platforms.
- Added [binary-only distributions][164] for FreeBSD `armv5`, `armv6`, and `armv7` and Linux `ppc64le` and `s390x`.
- Added the `is_silenced` Boolean attribute to the event.Check object to indicate whether the event was silenced at the time it was processed.

**IMPROVEMENTS:**

- ([Commercial feature][162]) Added support for the `memberOf` attribute in Active Directory (AD).
- ([Commercial feature][162]) Added more descriptive information for errors in the federated web UI.
- The `dead` and `handleUpdate` methods in keepalived now use `EntityConfig` and `EntityState` respectively.
- The `dead()` and `createProxyEntity()` methods in eventd now use `corev3.EntityConfig` and `corev3.EntityState`.
- Agent entity updates now ignore state-related fields.
- You can now manage Sensu agent configuration via the HTTP API.

> - Sensu now passes users' secondary groups (i.e. groups other than the Sensu user group) to `chroot` for sysvinit services, which gives the Sensu agent and backend access to the file access writes that are granted to the secondary groups.

Author: ```suggestion
- For sysvinit services, Sensu now passes users' secondary groups (i.e. groups other than the Sensu user group) to `chroot`, which gives the Sensu agent and backend access to the file access writes that are granted to the secondary groups.
```

---

## 6.0.0 release notes

> **July 23, 2020** &mdash; The latest release of Sensu Go, version 6.0.0, is now available for download.

Author: ```suggestion
**July 28, 2020** &mdash; The latest release of Sensu Go, version 6.0.0, is now available for download.
```

**PLACEHOLDER FOR RELEASE SYNOPSIS**

See the [upgrade guide][1] to upgrade Sensu to version 6.0.0.

**BREAKING CHANGES FOR SENSU 6.0:**

- The database schema for entities has changed.
As a result, after upgrading to Sensu 6.0, you will not be able to use your database with older versions of Sensu.
- In [binary-only distributions][164], the web UI is now a standalone product that is no longer distributed inside the `sensu-backend` binary.
See the [Sensu Go Web repository][163] for more information.
- After initial creation, you cannot change your `sensu-agent` entity configuration by modifying the agent's configuration file.

**NEW FEATURES:**

- ([Commercial feature][162]) Sensu now logs a warning when secrets cannot be sent to an agent because mTLS is not enabled.

> - ([Commercial feature][162]) Added JavaScript functions `sensu.CheckDependencies`, `sensu.EventStatus`, `sensu.FetchEvent`, and `sensu.ListEvents` to the filter execution environment so you can now query the Sensu event store for other events within the filter namespace.

Author: ```suggestion
- ([Commercial feature][162]) Added [JavaScript functions][169] `sensu.EventStatus`, `sensu.FetchEvent`, and `sensu.ListEvents` to the filter execution environment so you can now query the Sensu event store for other events within the filter namespace.
- ([Commercial feature][162]) Added [Sensu query expression][168] `sensu.CheckDependencies`.
```

---

## 6.0.0 release notes

**July 28, 2020** &mdash; The latest release of Sensu Go, version 6.0.0, is now available for download.

> **PLACEHOLDER FOR RELEASE SYNOPSIS**

Author: ```suggestion
**PLACEHOLDER FOR RELEASE SYNOPSIS**
With Sensu Go 6, you can control everything through the API. You can still use configuration management tools to bootstrap agent entities, but you don’t need to! Our new agent entity management feature via the backend configuration API nearly eliminates the need for external (or out-of-band) configuration management for Sensu, empowering you to manage agent entity subscriptions and automate the discovery of system facts without updating agent local configuration files. Run a sensuctl command, click a button in the web UI, or execute a custom check plugin.
```

See the [upgrade guide][1] to upgrade Sensu to version 6.0.0.

**BREAKING CHANGES FOR SENSU 6.0:**

- The database schema for entities has changed.
As a result, after upgrading to Sensu 6.0, you will not be able to use your database with older versions of Sensu.
- In [binary-only distributions][164], the web UI is now a standalone product that is no longer distributed inside the `sensu-backend` binary.
See the [Sensu Go Web repository][163] for more information.
- After initial creation, you cannot change your `sensu-agent` entity configuration by modifying the agent's configuration file.

**NEW FEATURES:**

- ([Commercial feature][162]) Sensu now logs a warning when secrets cannot be sent to an agent because mTLS is not enabled.
- ([Commercial feature][162]) Added [JavaScript functions][169] `sensu.EventStatus`, `sensu.FetchEvent`, and `sensu.ListEvents` to the filter execution environment so you can now query the Sensu event store for other events within the filter namespace.

> - ([Commercial feature][162]) Added [Sensu query expression][168] `sensu.CheckDependencies`.

Author: ```suggestion
```

- ([Commercial feature][162]) Docker images now set their own default values for environment variables `SENSU_AGENT_API_URL`, `SENSU_BACKEND_API_URL`, `SENSU_BACKEND_ETCD_INITIAL_CLUSTER`, `SENSU_BACKEND_ETCD_ADVERTISE_CLUSTER`, `SENSU_BACKEND_ETCD_INITIAL_ADVERTISE_PEER_URLS`, `SENSU_BACKEND_ETCD_LISTEN_CLIENT_URLS`, and `ETCD_LISTEN_PEER_URLS`.
- ([Commercial feature][162]) Added Linux packages for 386; armv5, armv6, and armv7; MIPS hard float, MIPS LE hard float, and MIPS 64 LE hard float; ppc64le; and s390x architectures.
See the [supported platforms][165] page for a complete list of Sensu’s supported platforms.

> - Added [binary-only distributions][164] for FreeBSD `armv5`, `armv6`, and `armv7` and Linux `ppc64le` and `s390x`.

Author: ```suggestion
- Added [Sensu query expression][168] `sensu.CheckDependencies`.
- Added [binary-only distributions][164] for FreeBSD `armv5`, `armv6`, and `armv7` and Linux `ppc64le` and `s390x`.
```

---

## 6.0.0 release notes

**July 28, 2020** &mdash; The latest release of Sensu Go, version 6.0.0, is now available for download.

**PLACEHOLDER FOR RELEASE SYNOPSIS**
With Sensu Go 6, you can control everything through the API. You can still use configuration management tools to bootstrap agent entities, but you don’t need to! Our new agent entity management feature via the backend configuration API nearly eliminates the need for external (or out-of-band) configuration management for Sensu, empowering you to manage agent entity subscriptions and automate the discovery of system facts without updating agent local configuration files. Run a sensuctl command, click a button in the web UI, or execute a custom check plugin.

> 

Author: This draft release synopsis is taken from [Sean's blog post](https://docs.google.com/document/d/1kMlU4CvBV2g4oPDpQWBVBXmy1M02ksBfaVItqSd1YQY/edit#) -- it will need some editing to make sure it's not a word-for-word duplicate

- The web UI is now a standalone product that is no longer distributed inside the `sensu-backend` binary. See the [Sensu Go Web repository][163] for more information.
- After initial creation, you cannot change your `sensu-agent` entity configuration by modifying the agent's configuration file.

**NEW FEATURES:**

- ([Commercial feature][162]) PLACEHOLDER.
- Added supported packages for Raspbian 7, 8, 9, and 10.
- Added binary-only distributions for FreeBSD `armv5`, `armv6`, and `armv7` and Linux `ppc64le` and `s390x`.
- Added `sensu.FetchEvent(entity, check)` and `sensu.ListEvents()` to the JavaScript filter execution environment, which means users can now query the Sensu event store for other events within the filter namespace.
- Sensu now logs a warning when secrets cannot be sent to an agent because mTLS is not enabled.

**IMPROVEMENTS:**

- ([Commercial feature][162]) Added support for the `memberOf` attribute in Active Directory (AD).
- ([Commercial feature][162]) Added more descriptive information for errors in the federated web UI.

> - ([Commercial feature][162]) Sensu now passes users' secondary groups (i.e. groups other than the Sensu user group) to `chroot` for sysvinit services, which gives the Sensu agent and backend access to the file access writes that are granted to the secondary groups.

Reviewer: I'm not 100% sure this falls under the "Commercial feature" umbrella: this applies to the packages we publish, but one does not need a license to make use of that change. This is a package improvement for some (older) platforms.

- ([Commercial feature][162]) CGO builds for Darwin `amd64` are now automated.

> - ([Commercial feature][162]) AppVeyor is now triggered directly from a CircleCI step rather than CMake.

Reviewer: I'm not sure we need to mention automatic CGO build for Darwin/amd64 and AppVeyor changes in the release notes. Those are changes to our internal process only as far as I understand.

- Output of `sensuctl asset add` now includes help for using the runtime asset.

[NEED TO ADD]
Improves logging around the agent websocket connection.
Entities are now stored as two separate data structures, in order to optimize data access patterns.
The dead and handleUpdate methods in keepalived now use EntityConfig and EntityState respectively.
The dead() and createProxyEntity() methods in eventd now use corev3.EntityConfig and corev3.EntityState.

> sensu-agent configuration can now be managed via the HTTP API.

Reviewer: This is one of the biggest, if not the biggest change that's coming with 6; so definitely need to add it :)

---

## 6.0.0 release notes

> **July 28, 2020** &mdash; The latest release of Sensu Go, version 6.0.0, is now available for download.

Author: ```suggestion
**August 10, 2020** &mdash; The latest release of Sensu Go, version 6.0.0, is now available for download.
```

See the [upgrade guide][1] to upgrade Sensu to version 6.0.0.

**BREAKING CHANGES FOR SENSU 6.0:**

- The database schema for entities has changed.
As a result, after upgrading to Sensu 6.0, you will not be able to use your database with older versions of Sensu.
- In [binary-only distributions][164], the web UI is now a standalone product that is no longer distributed inside the `sensu-backend` binary.
See the [Sensu Go Web repository][163] for more information.
- After initial creation, you cannot change your `sensu-agent` entity configuration by modifying the agent's configuration file.

**NEW FEATURES:**

- ([Commercial feature][162]) Sensu now logs a warning when secrets cannot be sent to an agent because mTLS is not enabled.
- ([Commercial feature][162]) Added [JavaScript functions][169] `sensu.EventStatus`, `sensu.FetchEvent`, and `sensu.ListEvents` to the filter execution environment so you can now query the Sensu event store for other events within the filter namespace.

> - ([Commercial feature][162]) Docker images now set their own default values for environment variables `SENSU_AGENT_API_URL`, `SENSU_BACKEND_API_URL`, `SENSU_BACKEND_ETCD_INITIAL_CLUSTER`, `SENSU_BACKEND_ETCD_ADVERTISE_CLUSTER`, `SENSU_BACKEND_ETCD_INITIAL_ADVERTISE_PEER_URLS`, `SENSU_BACKEND_ETCD_LISTEN_CLIENT_URLS`, and `ETCD_LISTEN_PEER_URLS`.

Author: ```suggestion
- ([Commercial feature][162]) Docker-only Sensu now binds to the hostname of containers instead of `localhost`. Docker images now set their own default values for environment variables `SENSU_AGENT_API_URL`, `SENSU_BACKEND_API_URL`, `SENSU_BACKEND_ETCD_INITIAL_CLUSTER`, `SENSU_BACKEND_ETCD_ADVERTISE_CLUSTER`, `SENSU_BACKEND_ETCD_INITIAL_ADVERTISE_PEER_URLS`, `SENSU_BACKEND_ETCD_LISTEN_CLIENT_URLS`, and `ETCD_LISTEN_PEER_URLS`.
```

**IMPROVEMENTS:**

- ([Commercial feature][162]) Added support for the `memberOf` attribute in Active Directory (AD).
- ([Commercial feature][162]) Added more descriptive information for errors in the federated web UI.
- The `dead` and `handleUpdate` methods in keepalived now use `EntityConfig` and `EntityState` respectively.
- The `dead()` and `createProxyEntity()` methods in eventd now use `corev3.EntityConfig` and `corev3.EntityState`.
- Agent entity updates now ignore state-related fields.
- You can now manage Sensu agent configuration via the HTTP API.
- For sysvinit services, Sensu now passes users' secondary groups (i.e. groups other than the Sensu user group) to `chroot`, which gives the Sensu agent and backend access to the file access writes that are granted to the secondary groups.
- Output of `sensuctl asset add` now includes help for using the runtime asset.
- For role bindings and cluster role bindings, [`subjects.name`][166] values can now include unicode characters, and [`roleRef.type`][167] and [`subjects.type`][166] values are now automatically capitalized.
- Improved logging for the agent websocket connection.

> - Improved the wording of the secret provider error message.

Author: ```suggestion
- Improved the wording of the secret provider error message.
- Upgraded Go version from 1.13.7 to 1.13.15.
- Upgraded etcd version from 3.3.17 to 3.3.22.
```

**IMPROVEMENTS:**

- ([Commercial feature][162]) Added support for the `memberOf` attribute in Active Directory (AD).
- ([Commercial feature][162]) Added more descriptive information for errors in the federated web UI.
- The `dead` and `handleUpdate` methods in keepalived now use `EntityConfig` and `EntityState` respectively.
- The `dead()` and `createProxyEntity()` methods in eventd now use `corev3.EntityConfig` and `corev3.EntityState`.
- Agent entity updates now ignore state-related fields.
- You can now manage Sensu agent configuration via the HTTP API.
- For sysvinit services, Sensu now passes users' secondary groups (i.e. groups other than the Sensu user group) to `chroot`, which gives the Sensu agent and backend access to the file access writes that are granted to the secondary groups.
- Output of `sensuctl asset add` now includes help for using the runtime asset.
- For role bindings and cluster role bindings, [`subjects.name`][166] values can now include unicode characters, and [`roleRef.type`][167] and [`subjects.type`][166] values are now automatically capitalized.
- Improved logging for the agent websocket connection.

> - Improved the wording of the secret provider error message.

Author: ```suggestion
- Improved the wording of the secret provider error message.
- Fewer keys in etcd are now stored for agents.
- Keepalive and round robin scheduling leases are now dealt with more efficiently.
```

- Upgraded Go version from 1.13.7 to 1.13.15.
- Upgraded etcd version from 3.3.17 to 3.3.22.

**FIXES:**

- ([Commercial feature][162]) Label selectors now work as expected with multiple requirements for events.
- ([Commercial feature][162]) Fixed an issue that prevented broken secrets providers from surfacing their errors.

> - Sensu now logs and returns and error if it cannot find a mutator.

Author: ```suggestion
- ([Commercial feature][162]) Fixed a bug for PostgreSQL datastores that could prevent GraphQL from retrieving all pages when fetching events in a namespace with more than 1000 total events, resulting in an unexpected error.
- Sensu now logs and returns and error if it cannot find a mutator.
```

- Errors produced in the agent by assets, check validation, token substitution, and event unmarshaling are logged once again.
- The User-Agent header is now set only upon on new client creation rather than upon each request.
- When the Sensu agent cannot parse the proper CA certificate path, Sensu logs this in the error message.
- Fixed a bug where highly concurrent event filtering could result in a panic.
- Fixed a bug where nil labels or annotations in an event filtering context would require you to explicitly check whether the annotations or labels are undefined.
With this fix, labels and annotations are always defined (although they may be empty).
- Fixed the log entry field for the check's name in schedulerd.

> - Fixed a bug for PostgreSQL datastores that could prevent GraphQL from retrieving all pages when fetching events in a namespace with more than 1000 total events, resulting in an unexpected error.

Author: ```suggestion
```

---

## 6.0.0 release notes

**August 10, 2020** &mdash; The latest release of Sensu Go, version 6.0.0, is now available for download.

> With Sensu Go 6.0.0, you can control everything through the API. You can still use configuration management tools to bootstrap agent entities, but you don’t need to! Our new agent entity management feature via the backend configuration API nearly eliminates the need for external (or out-of-band) configuration management for Sensu, which allows you to manage agent entity subscriptions and automate the discovery of system facts without updating agent local configuration files. Run a sensuctl command, click a button in the web UI, or execute a custom check plugin!

Author: ```suggestion

With Sensu Go 6.0.0, you can control everything through the API. You can still use configuration management tools to bootstrap agent entities, but you don’t need to! Our new agent entity management feature via the backend configuration API nearly eliminates the need for external (or out-of-band) configuration management for Sensu, which allows you to manage agent entity subscriptions and automate the discovery of system facts without updating agent local configuration files. Run a sensuctl command, click a button in the web UI, or execute a custom check plugin!
```

- You can now manage Sensu agent configuration via the HTTP API.
- For sysvinit services, Sensu now passes users' secondary groups (i.e. groups other than the Sensu user group) to `chroot`, which gives the Sensu agent and backend access to the file access writes that are granted to the secondary groups.
- Output of `sensuctl asset add` now includes help for using the runtime asset.
- For role bindings and cluster role bindings, [`subjects.name`][166] values can now include unicode characters, and [`roleRef.type`][167] and [`subjects.type`][166] values are now automatically capitalized.
- Improved logging for the agent websocket connection.
- Improved the wording of the secret provider error message.
- Fewer keys in etcd are now stored for agents.
- Keepalive and round robin scheduling leases are now dealt with more efficiently.
- Upgraded Go version from 1.13.7 to 1.13.15.
- Upgraded etcd version from 3.3.17 to 3.3.22.

**FIXES:**

- ([Commercial feature][162]) Label selectors now work as expected with multiple requirements for events.
- ([Commercial feature][162]) Fixed an issue that prevented broken secrets providers from surfacing their errors.

> - ([Commercial feature][162]) Fixed a bug for PostgreSQL datastores that could prevent GraphQL from retrieving all pages when fetching events in a namespace with more than 1000 total events, resulting in an unexpected error.

Author: ```suggestion
- ([Commercial feature][162]) Fixed a bug for PostgreSQL datastores that could prevent GraphQL from retrieving all pages when fetching events in a namespace with more than 1000 total events, resulting in an unexpected error.
- ([Commercial feature][162]) Fixed a bug that could cause the backend to panic in case of PostgreSQL errors.
```

---

## 6.0.0 release notes

**August 10, 2020** &mdash; The latest release of Sensu Go, version 6.0.0, is now available for download.

> 
> **PLACEHOLDER FOR RELEASE SYNOPSIS**

Author: ```suggestion
```

> With Sensu Go 6, you can control everything through the API. You can still use configuration management tools to bootstrap agent entities, but you don’t need to! Our new agent entity management feature via the backend configuration API nearly eliminates the need for external (or out-of-band) configuration management for Sensu, empowering you to manage agent entity subscriptions and automate the discovery of system facts without updating agent local configuration files. Run a sensuctl command, click a button in the web UI, or execute a custom check plugin.

Author: ```suggestion
With Sensu Go 6.0.0, you can control everything through the API. You can still use configuration management tools to bootstrap agent entities, but you don’t need to! Our new agent entity management feature via the backend configuration API nearly eliminates the need for external (or out-of-band) configuration management for Sensu, which allows you to manage agent entity subscriptions and automate the discovery of system facts without updating agent local configuration files. Run a sensuctl command, click a button in the web UI, or execute a custom check plugin!
```

[158]: /sensu-go/5.21/commercial/
[160]: /sensu-go/5.21/reference/backend#fips-openssl
[161]: /sensu-go/5.21/reference/agent#fips-openssl
[162]: /sensu-go/6.0/commercial/
[163]: https://github.com/sensu/web
[164]: /sensu-go/6.0/platforms/#binary-only-distributions
[165]: /sensu-go/6.0/platforms/
[166]: /sensu-go/6.0/reference/rbac/#subjects-specification

> [167]: /sensu-go/6.0/reference/rbac/#roleref-specification

Author: ```suggestion
[167]: /sensu-go/6.0/reference/rbac/#roleref-specification
[168]: /sensu-go/6.0/reference/sensu-query-expressions/#sensucheckdependencies
[169]: /sensu-go/6.0/reference/filters/#build-event-filter-expressions-with-javascript-execution-functions
```
